atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImplTest.java

```diff
@@ -82,48 +89,92 @@ public void lockEventInvalidatesValue() {
         verify(metrics).decreaseCacheSize(EXPECTED_SIZE);
     }
 
+    @Test
+    public void lockEventInvalidatesValueForRowLevelReference() {
+        valueStore.applyEvent(WATCH_EVENTS);
+        valueStore.putValue(ROW_LEVEL_CELL_REFERENCE, VALUE_1);
+        int expectedSize = EntryWeigher.INSTANCE.weigh(ROW_LEVEL_CELL_REFERENCE, 1);
+
+        verify(metrics).increaseCacheSize(expectedSize);
+
+        assertExpectedValue(EXACT_ROW_TABLE, ROW_LEVEL_CELL, CacheEntry.unlocked(VALUE_1));
+
+        valueStore.applyEvent(ROW_LOCK_EVENT);
+        assertExpectedValue(EXACT_ROW_TABLE, ROW_LEVEL_CELL, CacheEntry.locked());
+
+        verify(metrics).decreaseCacheSize(expectedSize);
+    }
+
     @Test
     public void unlockEventsClearLockedEntries() {
```

Comment:
For the lock event test, I added a parallel for row-level lock events.
Here, I test both row-level and table-level unlocks in the same test.

Which style is preferred?